internal/resources/metal/vrf/datasource.go

```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func dataSourceMetalVRF() *schema.Resource {
+func DataSource() *schema.Resource {
```

Comment:
While we're in here, let's add a test of the data source in `datasource_test.go` (I realize it shares almost all of its code with the resource, but that may not always be the case, so having a test helps protect against unintended consequences later).